invenio_records_resources/services/files/components/metadata.py

```diff
@@ -34,8 +46,4 @@ def update_file_metadata(self, identity, id, file_key, record, data):
     # TODO: `commit_file` might vary based on your storage backend (e.g. S3)
     def commit_file(self, identity, id, file_key, record):
         """Commit file handler."""
-        # TODO: Add other checks here (e.g. verify checksum, S3 upload)
-        file_obj = ObjectVersion.get(record.bucket.id, file_key)
-        if not file_obj:
-            raise Exception(f"File with key {file_key} not uploaded yet.")
-        record.files[file_key] = file_obj
+        record.files.commit(file_key)
```

Comment:
I'm still in favour of delegating the final commit to the manager, because otherwise we would have to use `ObjectVersions` and other data layer objects here. On the other hand, I also agree that it should pass by the `Transfer` object in order to make the behaviour configurable and storage-class dependant. I've refactored it to reflect this. (still pending to see what to do with `files.update`, all the other commands follow `service -> cmp -> transfer -> manager`